charmhelpers/contrib/hahelpers/apache.py

```diff
@@ -61,6 +68,11 @@ def get_cert(cn=None):
                 if not key:
                     key = relation_get(ssl_key_attr,
                                        rid=r_id, unit=unit)
+
+    # this likely fails too quietly
```

Comment:
Should it have logging, then?

Reply:
What's meant by "too quietly" applies to code's logic - there could be a case that false-value these returns could be simply used in conditions deteremining whether to configure TLS at all, which poses a huge red flag for the existing codebase.

All functions called by `x509_validate_cert` to validate parity and trust chains already have arguably sufficient logging about why things may have been going awry.